Moq.AutoMock/AutoMocker.cs

```diff
@@ -436,6 +436,41 @@ public ISetup<TService> Setup<TService>(Expression<Action<TService>> setup)
             return Setup<ISetup<TService, TReturn>, TService>(m => m.Setup(setup));
         }
 
+        /// <summary>
+        /// Specifies a setup on the mocked type for a call to a void method.
+        /// All parameters are filled with <see cref ="It.IsAny" /> according to the parameter's type.
+        /// </summary>
+        /// <remarks>
+        /// This may only be used on methods that are not overloaded.
+        /// This will create the mock when necessary.
+        /// </remarks>
+        /// <typeparam name="TService">The service type</typeparam>
+        /// <param name="methodName">The name of the expected method invocation.</param>
+        /// <returns></returns>
+        public ISetup<TService> SetupWithAny<TService>(string methodName)

```

Comment:
Might be worth [documenting](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/exception) these with the applicable exceptions, mainly to help drive home the point of only using on methods that aren't overloaded.

Reply:
That is a good idea, I can do that.